libtenzir/src/tql2/eval_impl.cpp

```diff
@@ -152,6 +154,48 @@ auto evaluator::eval(const ast::root_field& x) -> series {
   return null();
 }
 
+auto evaluator::eval(const ast::index_expr& x) -> series {
+  auto value = eval(x.expr);
+  auto index = eval(x.index);
+  if (auto number = index.as<int64_type>()) {
+    auto list = value.as<list_type>();
+    if (not list) {
+      diagnostic::warning("cannot index into `{}` with `{}`", value.type.kind(),
+                          index.type.kind())
+        .primary(x.index)
+        .emit(ctx_);
+      return null();
+    }
+    auto list_values = list->array->values();
+    auto value_type = list->type.value_type();
+    auto b = value_type.make_arrow_builder(arrow::default_memory_pool());
+    check(b->Reserve(list->length()));
+    auto out_of_bounds = false;
+    for (auto i = int64_t{0}; i < list->length(); ++i) {
+      auto target = number->array->Value(i);
+      auto length = list->array->value_length(i);
+      if (target < 0) {
+        target = length + target;
+      }
+      if (target >= length) {
```

Comment:
```suggestion
      if (target < 0 or target >= length) {
```

This can be a quick fix?

Reply:
Yep! That's exactly what I just did.